test/test_nn.py

```diff
@@ -740,6 +740,34 @@ def test_no_grad(self):
                 self.assertFalse(output2.requires_grad)
                 self.assertRaises(RuntimeError, lambda: output2.backward(torch.ones(1, 5, 10, 10)))
 
```

Comment:
Need to add `device` and `dtype` for the test case, otherwise it will only test on CPU.